# onekeyBulkSeq

onekeyBulkSeq  is a collection of scripts for QTL-seq and MutMap using NGS Bulk Segregant Analysis.

Mutmap scripts support analysis using one wildtype parent and one mutant pool.

QTLseq scripts support analysis using two parents(mutant and wildtype) and two pools.

## Installation
### Dependencies
#### Softwares
- [vcftools](https://vcftools.github.io/man_latest.html)
- [R](https://www.r-project.org/)
- [perl](https://www.perl.org/)

#### R packages
- ggplot2
- reshape2

### Installation dependencies using bioconda

```
conda install r-ggplot2 r-reshape2 vcftools -c bioconda
```

## Usage
### run MutMap

```
perl mutmap/mutmap.pl -vcf data/mutmapDemo.vcf  -parent P -bulk Bulk -key Demo -od mutmapOut -win 2000000 -step 100000 -minCount 5 -ems
```

```
Options:
-vcf        <file>      required,  vcf file path
-parent     <string>    required, wild type parent name in vcf file
-bulk       <string>    required, mutant Pool name in vcf file
-key        <string>    optional,  output mutmap 
-od         <string>    optional, output directory,default:MutmapOut

-minDP1     <int>       optional, minimum depth for parent sample，default:8
-minDP2     <int>       optional, maximum depth for pool sample，default:20
-maxDP1     <int>       optional, minimum depth for parent sample, default:100
-maxDP2     <int>       optional, maximum depth for pool sample，default 300
-minCount   <int>       optional, minimum snp count required for window sliding ，default 5

-win        <int>       optional, sliding window size, default 2000000
-step       <int>       optional, sliding window step，default 100000
-ems        <option>    optional， if mutant is produced by EMS.

-thr        <float>     optional， SNP-index threshold for plotting.
-help       help
```


## run QTLseq


```
perl QTLseq/QTLseq.pl   -vcf data/demo.vcf -parent1 P  -parent2 M -bulk1 B1 -bulk2 B2 -method both -key Demo  -od BSAoutput
```

```
Options:
-vcf        <file>      required,  vcf file path
-parent1    <string>    required, mutant parent name in vcf file
-parent2    <string>    required, wild type parent name in vcf file
-bulk1      <string>    required, mutant Pool name in vcf file
-bulk2      <string>    required, wild type Pool name in vcf file

-key        <string>    optional,  output BSA 
-od         <string>    optional, output directory,default:BSAOut

-minDP1     <int>       optional, minimum depth for parent sample，default:8
-minDP2     <int>       optional, maximum depth for pool sample，default:20
-maxDP1     <int>       optional, minimum depth for parent sample, default:100
-maxDP2     <int>       optional, maximum depth for pool sample，default 300
-minCount   <int>       optional, minimum snp count required for window sliding ，default 5

-win        <int>       optional, sliding window size, default 2000000
-step       <int>       optional, sliding window step，default 100000
-power      <int>       optional， ED power , default 4

-plotthrED     <float>     optional， ED threshold for plotting, 
                           default value is calculated automatically by Median+3*SD.
						   
-plotthrIndex  <float>     optional， SNP-index threshold for plotting.
                           default value is generated by simulation.
						   
-help       help
```



## Citation

- Akira Abe, Shunichi Kosugi, Kentaro Yoshida, Satoshi Natsume, Hiroki Takagi, Hiroyuki Kanzaki, Hideo Matsumura, Kakoto Yoshida, Chikako Mitsuoka, Muluneh Tamiru, Hideki Innan, Liliana Cano, Sophien Kamoun & Ryohei Terauchi (2012). Genome sequencing reveals agronomically important loci in rice using MutMap. Nature Biotechnol. 30:174-179. [[URL]](https://www.nature.com/articles/nbt.2095)
- Hiroki Takagi, Akira Abe, Kentaro Yoshida, Shunichi Kosugi, Satoshi Natsume, Chikako Mitsuoka, Aiko Uemura, Hiroe Utsushi, Muluneh Tamiru, Shohei Takuno, Hideki Innan, Liliana M. Cano, Sophien Kamoun, Ryohei Terauchi (2013).  QTL-seq: rapid mapping of quantitative trait loci in rice by whole genome resequencing of DNA from two bulked populations. Plant journal 74:174-183. [[URL]](https://doi.org/10.1111/tpj.12105)
